src/server/bg_services/replication_log_parser.js

```diff
@@ -192,7 +192,8 @@ function create_candidates(logs) {
         const logs_for_key = Array.from(logs_for_key_time_map.values());
         const sorted_logs_for_key = logs_for_key.sort((a, b) => a.time.getTime() - b.time.getTime());
 
-        candidates[key] = sorted_logs_for_key;
+        // Storing only latest log per key in candidates
+        candidates[key] = [sorted_logs_for_key[sorted_logs_for_key.length - 1]];
```

Comment:
If we just care about the log entry with maximum timestamp then maybe we shouldn't be sorting them at all, just finding the log entry with maximum timestamp should do?

Reply:
Yes, that will be fine.